cmd/snap/cmd_run.go

```diff
@@ -482,29 +473,103 @@ func (x *cmdRun) straceOpts() (opts []string, raw bool, err error) {
 	return opts, raw, nil
 }
 
+func getInfoAndApp(snapName, appName string, rev snap.Revision) (*snap.Info, *snap.AppInfo, error) {
+	info, err := getSnapInfo(snapName, rev)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	app := info.Apps[appName]
+	if app == nil {
+		return nil, nil, fmt.Errorf(i18n.G("cannot find app %q in %q"), appName, snapName)
+	}
+
+	return info, app, nil
+}
+
+func maybeWaitWhileInhibited(snapName string, appName string) (info *snap.Info, app *snap.AppInfo, hintFlock *osutil.FileLock, err error) {
+	if features.RefreshAppAwareness.IsEnabled() {
+		flow := newInhibitionFlow(snapName)
```

Comment:
In particular, it seems like it could be re-created on every invocation of `inhibited()`, and even then only if `!notified` (assuming you only want to call it once, which I'm not sure about).